swssconfig/swssconfig.cpp

```diff
@@ -39,12 +43,35 @@ void dump_db_item(KeyOpFieldsValuesTuple &db_item)
     SWSS_LOG_DEBUG("]");
 }
 
-bool write_db_data(vector<KeyOpFieldsValuesTuple> &db_items)
+shared_ptr<ProducerStateTable> get_table(unordered_map<string, shared_ptr<ProducerStateTable>>& table_map, RedisPipeline &pipeline, string table_name, set<string>  zmq_tables, std::shared_ptr<ZmqClient> zmq_client)
+{
+    shared_ptr<ProducerStateTable> p_table= nullptr;
+    auto findResult = table_map.find(table_name);
+    if (findResult == table_map.end())
+    {
+        if ((zmq_tables.find(table_name) != zmq_tables.end()) && (zmq_client != nullptr)) {
+            p_table = make_shared<ZmqProducerStateTable>(&pipeline, table_name, *zmq_client, true);
+        }
+        else {
+            p_table = make_shared<ProducerStateTable>(&pipeline, table_name, true);
+        }
+
+        table_map.emplace(table_name, p_table);
+    }
+    else
+    {
+        p_table = findResult->second;
+    }
+
+    return p_table;
+}
+
+bool write_db_data(vector<KeyOpFieldsValuesTuple> &db_items, set<string>  zmq_tables, std::shared_ptr<ZmqClient> zmq_client)
```

Comment:
>set [](http://example.com/codeflow?start=61&length=3)

Set copy is heavy. Do you want to use const reference?

Reply:
Fixed